api/src/commonMain/kotlin/dev/kord/cache/api/observables/Cache.kt

```diff
@@ -0,0 +1,33 @@
+package dev.kord.cache.api.observables
+
+interface Cache<T: Any> {
+
+    /**
+     * Removes any cached value [T] that satisfies the given [transform] function.
+     * @param transform A function that takes a value of type [T] and returns a boolean.
+     */
+
+    /**
+     * Returns the first value that satisfies the given [transform] function, or null if none is found.
+     * @param transform A function that takes a value of type [T] and returns a boolean.
+     * @return The first value that satisfies the [transform] function, or null if none is found.
+     */
+    public suspend fun firstOrNull(transform: (T) -> Boolean): T?
+
+
+    public suspend fun removeIf(transform: (T) -> Boolean)
```

Comment:
```suggestion
    public suspend fun firstOrNull(predicate: (T) -> Boolean): T?


    public suspend fun removeIf(predicate: (T) -> Boolean)
```
i think `predicate` is a better name and also commonly used in e.g. stdlib